File: game/forest/forest_utils.py
from game_data import GameData


class ForestUtils:

    @staticmethod
    def reduce_lives(context: GameData, obstacle_type: int = 0):
        context.forest_lives -= 1

        # Track which obstacle caused this life loss
        if context.forest_obstacles_hit is None:
            context.forest_obstacles_hit = []
        if obstacle_type > 0:
            context.forest_obstacles_hit.append(obstacle_type)

        if context.forest_lives == 0:
            from forest.talking_game_over import TalkingGameOver
            return TalkingGameOver
        else:
            from forest.talking_after_hurt import TalkingAfterHurt
            return TalkingAfterHurt

File: game-web/src/forest/forestUtils.ts
/** Port of game/forest/forest_utils.py */

import type { GameData } from '../gameData';
import type { StateClass } from '../state';
import { TalkingAfterHurt } from './talkingAfterHurt';
import { TalkingGameOver } from './talkingGameOver';

export class ForestUtils {
  static reduceLives(context: GameData, obstacleType: number = 0): StateClass {
    context.forestLives -= 1;

    if (obstacleType > 0) {
      context.forestObstaclesHit.push(String(obstacleType));
    }

    if (context.forestLives === 0) {
      return TalkingGameOver;
    } else {
      return TalkingAfterHurt;
    }
  }
}
